### ISSUE TITLE
Miliseconds?

### ISSUE THREAD

**Issue by alex-calamar** [danomatika/ofxMidi#67]
Good evening...
Sorry, maybe I'm wrong or maybe this issue has been already discussed around here (I couldn't find it, if so)...
In all the conversions (at least in the ofxMidiTimeCode.cpp file; haven't checked others because I was dealing with this one, sorry) from and to miliseconds you multiply and divide seconds by 100... Shouldn't it be divided or multiplied by 1000?
I'm new in developing in here and I beg you pardon in advance if I'm wrong... But it seemed strange to me and I couldn't find a reason... Perhaps you are so kind of explaining me here.
Thanks in advance for your kind time.
Alex

**Comment by alex-calamar** [comment]
Mmm... Sorry again, maybe I'm not understanding your code... Forgive a beginner...
What about this?

> /// convert to time in seconds
> double ofxMidiTimecodeFrame::toSeconds() const {
> 	double time = (double)hours * 3600.0; // 60.0 * 60.0
> 	time += (double)minutes * 60.0;
> 	time += (double)seconds;
> 	time += ofxMidiTimecode::framesToMs(frames, rate) / **100**;
> 	return time;
> }

Thanks for your time.
Alex

**Comment by alex-calamar** [comment]
Yes, with your code I had some strange behaviours... Then I reviewed and saw this "100 - 1000" issue... I changed your conversions from 100 to 1000 and then it worked fine for me...

Anyway it is strange that you have conversions from seconds to milliseconds just multiplying by 100... Like these functions here... Don't you think?

```
// -----------------------------------------------------------------------------
int ofxMidiTimecode::framesToMs(int frames, unsigned char rate) {
	return (int)((1.0/rateToFps(rate)) * (double)frames * 100.0);
}

// -----------------------------------------------------------------------------
int ofxMidiTimecode::msToFrames(int ms, unsigned char rate) {
	return (int)((double)ms / (1.0/rateToFps(rate)) / 100.0);
}
```

Well, thanks again for your time... Just want to help.

Regards.
Alex

**Comment by danomatika** [comment]
Those look questionable, yeah. Also, no need to apologize when you think you found a bug!

**Comment by danomatika** [comment]
Try changing those conversions to 1000 and let me know if it works better overall.

**Comment by alex-calamar** [comment]
Yes, you are right... I thought that maybe you could do some kind of rounding and return just "seconds", but I see that toSeconds returns a float, so it can be easily multiplied by 1000 to get miliseconds, you are right.

Thanks then.
Nice to feel that I helped to a project in here.
See you around.
Alex

**Comment by danomatika** [comment]
>Nice to feel that I helped to a project in here.

Sure. Every little bit helps. I actually ported the MTC and MIDI Clock code from another project and I noticed I had made the 100 / 1000 mistake there originally. So this issue has fixed a couple projects now.
